lib/elixir/lib/calendar/duration.ex

```diff
@@ -0,0 +1,222 @@
+      %Duration{week: 2, day: 3}
+
+  """
+  @spec add(t, t) :: t
+  def add(%Duration{} = d1, %Duration{} = d2) do
+    %Duration{
+      year: d1.year + d2.year,
+      month: d1.month + d2.month,
+      week: d1.week + d2.week,
+      day: d1.day + d2.day,
+      hour: d1.hour + d2.hour,
+      minute: d1.minute + d2.minute,
+      second: d1.second + d2.second,
+      microsecond: d1.microsecond + d2.microsecond
+    }
+  end
+
+  @doc """
+  Subtracts two durations.
+
+  ## Examples
+
+      iex> Duration.subtract(%Duration{week: 2, day: 1}, %Duration{day: 2})
+      %Duration{week: 2, day: -1}
+
+  """
+  @spec subtract(t, t) :: t
+  def subtract(%Duration{} = d1, %Duration{} = d2) do
+    %Duration{
+      year: d1.year - d2.year,
+      month: d1.month - d2.month,
+      week: d1.week - d2.week,
+      day: d1.day - d2.day,
+      hour: d1.hour - d2.hour,
+      minute: d1.minute - d2.minute,
+      second: d1.second - d2.second,
+      microsecond: d1.microsecond - d2.microsecond
+    }
+  end
+
+  @doc """
+  Multiplies all duration units by given integer.
+
+  ## Examples
+
+      iex> Duration.multiply(%Duration{day: 1, minute: 15, second: -10}, 3)
+      %Duration{day: 3, minute: 45, second: -30}
+
+  """
+  @spec multiply(t, integer) :: t
+  def multiply(%Duration{} = duration, integer) when is_integer(integer) do
+    %Duration{
+      year: duration.year * integer,
+      month: duration.month * integer,
+      week: duration.week * integer,
+      day: duration.day * integer,
+      hour: duration.hour * integer,
+      minute: duration.minute * integer,
+      second: duration.second * integer,
+      microsecond: duration.microsecond * integer
+    }
+  end
+
+  @doc """
+  Negates all duration units.
+
+  ## Examples
+
+      iex> Duration.negate(%Duration{day: 1, minute: 15, second: -10})
+      %Duration{day: -1, minute: -15, second: 10}
+
+  """
+  @spec negate(t) :: t
+  def negate(%Duration{} = duration) do
+    %Duration{
+      year: -duration.year,
+      month: -duration.month,
+      week: -duration.week,
+      day: -duration.day,
+      hour: -duration.hour,
+      minute: -duration.minute,
+      second: -duration.second,
+      microsecond: -duration.microsecond
+    }
+  end
+
+  @doc """
+  Compares two durations.
+
+  Returns `:gt` if the first duration is longer than the second and `:lt` for vice versa.
+  If the two durations are equal in length in seconds `:eq` is returned.
+
+  ## Examples
+
+      iex> Duration.compare(%Duration{hour: 1, minute: 15}, %Duration{hour: 2, minute: -45})
+      :eq
+      iex> Duration.compare(%Duration{year: 1, minute: 15}, %Duration{minute: 15})
+      :gt
+      iex> Duration.compare(%Duration{day: 1, minute: 15}, %Duration{day: 2})
+      :lt
+      iex> Duration.compare(%Duration{day: 1, microsecond: 15}, %Duration{day: 1})
+      :gt
+
+  """
+  @spec compare(t, t, Calendar.calendar()) :: :lt | :eq | :gt
+  def compare(
```

Comment:
I would prefer to not have this function and try to keep duration calendar free (for now). We can always add it later.

Reply:
Understood, i got inspired by https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#compareTo-java.time.Duration- and thought it'd fit well with the other calendar modules. But i agree, we can always add it later 👍